File: src/main/java/com/portofino/realtrainmodunofficial/rail/util/RailMapSwitch.java
package com.portofino.realtrainmodunofficial.rail.util;

import net.minecraft.world.level.Level;

public final class RailMapSwitch extends RailMapBasic {
    public final RailDir startDir;
    public final RailDir endDir;
    private boolean open;

    public RailMapSwitch(RailPosition start, RailPosition end, RailDir startDir, RailDir endDir) {
        super(start, end);
        this.startDir = startDir;
        this.endDir = endDir;
    }

    public RailMapSwitch setState(boolean open) {
        this.open = open;
        return this;
    }

    public boolean isOpen() {
        return open;
    }

    public boolean isGettingPowered(Level level) {
        return super.isGettingPowered(level);
    }
}

File: src/main/kotlin/cc/mirukuneko/realtrainmodrenewed/rail/util/RailMapSwitch.kt
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright © 2026 mirukuneko and RealTrainModRenewed contributors
package cc.mirukuneko.realtrainmodrenewed.rail.util

import net.minecraft.world.level.Level

class RailMapSwitch(
    start: RailPosition,
    end: RailPosition,
    @JvmField val startDir: RailDir,
    @JvmField val endDir: RailDir
) : RailMapBasic(start, end) {

    var open: Boolean = false
        private set

    fun setState(open: Boolean): RailMapSwitch {
        this.open = open
        return this
    }

    fun isOpen(): Boolean = open

    // isGettingPowered inherited from RailMapBasic, cannot override final member
}
